### PR TITLE
(CDAP-17645) Applies SSH config from the system and default a timeout

### PR DESCRIPTION
**Pull request by chtyim**

### CODE REVIEW

**Review by CuriousVini — APPROVED** [review]

**Review by rmstar — APPROVED** [review]

File: cdap-common/src/main/java/io/cdap/cdap/common/ssh/DefaultSSHSession.java [hunk 1]
```diff
@@ -349,4 +365,27 @@ private int resolvePermissions(Set<PosixFilePermission> posixPermissions) {
...
+    // Detect and try to set configurations via OpenSSH configurations
+    for (Path configPath : Arrays.asList(Paths.get(System.getProperty("user.home"), ".ssh", "config"),
+                                         Paths.get(File.separator, "etc", "ssh", "config"))) {
+      if (Files.exists(configPath)) {
+        try {
+          OpenSSHConfig config = OpenSSHConfig.parseFile(configPath.toAbsolutePath().toString());
+          jSch.setConfigRepository(config);
+          break;
+        } catch (IOException e) {
+          LOG.debug("Ignore applying SSH config from file {} due to failure", configPath, e);
```
Comment (tivv) [inline cdap-common/src/main/java/io/cdap/cdap/common/ssh/DefaultSSHSession.java]: Should it be a warn or info?

**Review by tivv — APPROVED** [review]

File: cdap-common/src/main/java/io/cdap/cdap/common/ssh/DefaultSSHSession.java (hunk 1 above)
Comment (chtyim) [reply cdap-common/src/main/java/io/cdap/cdap/common/ssh/DefaultSSHSession.java]: Changed to warn

**Pull request merged**